dpypelines/s3_tar_received.py

```diff
@@ -1,22 +1,75 @@
+import importlib
+import json
+import os
+
+from dpytools.stores.directory.local import LocalDirectoryStore
+from dpypelines.pipeline.shared import notification as notify
+from dpypelines.pipeline.shared.schemas import get_config_schema_path
+from dpytools.s3.basic import decompress_s3_tar
+from dpypelines.pipeline.dataset_ingress_v1 import dataset_ingress_v1
+from dpytools.validation.json import validation
+from dpypelines.pipeline.shared import message
+
 def start(s3_object_name: str):
     """
     Handles the required behaviour when recieving a tar file indicated
     by an s3 object name.
 
-    Example s3_object_name: my-bucket/my-data.tar
+    Args:
+        s3_object_name (str): The S3 object name of the tar file to be processed.
+        Example: 'my-bucket/my-data.tar'
+
     """
 
-    # IN ALL CASES, the FOLLOWING STEPS WILL BE IN TRY CATCH
-    # BLOCKS AND WILL NOTIFY RELEVANT PARTIES IN EVENT OF FAILURE
+    # Decompress the tar file to the workspace
+    try:
+        decompress_s3_tar(s3_object_name, "input")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to decompress tar file {s3_object_name}", err))
+        raise err
+    
+    # Create a local directory store using the decompressed files
+    try:
+        localStore = LocalDirectoryStore(f"inputs")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to create local directory store at inputs", err))
+        raise err
 
-    # decompress tar file to workspace
+    # Check for the existence of a configuration file
+    try:
+        if not localStore.has_lone_file_matching("pipeline-config.json"):
+            msg = message.expected_local_file_missing("Missing pipeline-config.json", "pipeline-config.json", "start")
```

Comment:
can you just use `message.unexpected_error()`, this message doesn't quite match up as I initially thought it would. We'll have a task to look at this coming up.